application/models/services/Quotas.php

```diff
@@ -384,4 +392,212 @@ public function checkActionPermissions($action)
+            if (empty($aQuotas)) {
+                return $aMatchedQuotas;
+            }
+
+            // OK, we have some quota, then find if this $_SESSION have some set
+            // foreach ($aQuotasInfos as $aQuotaInfo)
+            foreach ($aQuotas as $oQuota) {
+                // if(!$aQuotaInfo['active']) {
+                if (!$oQuota->active) {
+                    continue;
+                }
+                // if(count($aQuotaInfo['members'])===0) {
+                if (count($oQuota->quotaMembers) === 0) {
+                    continue;
+                }
+                $iMatchedAnswers = 0;
+                $bPostedField = false;
+
+                ////Create filtering
+                // Array of field with quota array value
+                $aQuotaFields = [];
+                // Array of fieldnames with relevance value : EM fill $_SESSION with default value even is irrelevant (em_manager_helper line 6548)
+                $aQuotaRelevantFieldnames = [];
+                // To count number of hidden questions
+                $aQuotaQid = [];
+                //Fill the necessary filter arrays
+                foreach ($oQuota->quotaMembers as $oQuotaMember) {
+                    $aQuotaMember = $oQuotaMember->getMemberInfo();
+                    $aQuotaFields[$aQuotaMember['fieldname']][] = $aQuotaMember['value'];
+                    $aQuotaRelevantFieldnames[$aQuotaMember['fieldname']] = isset($_SESSION['survey_' . $surveyid]['relevanceStatus'][$aQuotaMember['qid']]) && $_SESSION['survey_' . $surveyid]['relevanceStatus'][$aQuotaMember['qid']];
+                    $aQuotaQid[] = $aQuotaMember['qid'];
+                }
+                $aQuotaQid = array_unique($aQuotaQid);
+
+                // Filter
+                // For each field : test if actual responses is in quota (and is relevant)
+                foreach ($aQuotaFields as $sFieldName => $aValues) {
+                    $bInQuota = isset($_SESSION['survey_' . $surveyid][$sFieldName])
+                        && in_array($_SESSION['survey_' . $surveyid][$sFieldName], $aValues);
+                    if ($bInQuota && $aQuotaRelevantFieldnames[$sFieldName]) {
+                        $iMatchedAnswers++;
+                    }
+                    if (!is_null(App()->request->getPost($sFieldName))) {
+                        // Need only one posted value
+                        $bPostedField = true;
+                        $aPostedQuotaFields[$sFieldName] = App()->getRequest()->getPost($sFieldName);
+                    }
+                }
+
+                ################ QUOTA ACTIONS START ################
+                $quotaAction = (int)$oQuota->action;
+                $actionBasedTriggerCondition = false;
+
+                if ($quotaAction === Quota::TERMINATE_AFTER_ALL_VISIBLE_QUOTA_QUESTION) {
+                    // check if all question that are part of the quotaMembers are hidden NOTE: this is kept for legacy compatibility
```

Comment:
Why for left compatibility ? It's a specific behaviour.

Reply:
because a lot of people have based there surveys on this behaviour, intended or not

Reply:
Yep, but the comment are here in future. It's not _kept for legacy compatibility_ it's one behaviour